src/App.js

```diff
@@ -1,26 +1,171 @@
+        style={completed ? completeStyle : null}
+      >{todo}</p>
+      <button className="Deleter" onClick={e => deleteTodo(todoId)}>X</button>
+    </div>
+  );
+}
+
+
+function TodoList({todos, toggleTodo, deleteTodo}) {
+  const listOfTodos = (
+    todos.map(todo =>  (
+     <TodoBox
+       key={todo.id}
+       todoId={todo.id}
+       todo={todo.todoItem}
+       completed={todo.completed}
+       deleteTodo={deleteTodo} 
+       toggleTodo={toggleTodo}
+     />
+    ))
+  );
+  return (
+    listOfTodos
+  );
+}
+
+
+function Footer({todos, changeViewFilter}) {
+  function handleClick(e) {
+    changeViewFilter(e.target.name);
+  }
+  
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div className="Footer">
+      <p>{todos.length} {todos.length === 1 ? 'todo' : 'total todos' }</p>
+      <button name="SHOW_ALL" onClick={handleClick}>ALL</button>
+      <button name="SHOW_ACTIVE" onClick={handleClick}>ACTIVE</button>
+      <button name="SHOW_COMPLETED" onClick={handleClick}>COMPLETED</button>
     </div>
   );
 }
 
-export default App;
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      viewFilter: 'SHOW_ALL',
+      todos: [
+        {
+          id: 0,
+          todoItem: 'Buy biscuits',
+          completed: true
+        },
+        {
+          id: 1,
+          todoItem: 'Walk the dog',
+          completed: false
+        },
+        {
+          id: 2,
+          todoItem: 'Watch movie',
+          completed: false
+        },
+        {
+          id: 3,
+          todoItem: 'Practice web dev',
+          completed: true
+        },
+      ],
+      nextId: 4
+    };
+  }
+  
+  handleNewTodo = newTodo => {
+    console.log(`Added "${newTodo}" to the todo list!`);
+    const newTodoObj = {id: this.state.nextId, todoItem: newTodo, completed: false};
+    const todos = [...this.state.todos, newTodoObj];
+    const newNextId = this.state.nextId + 1;
+    const newState = {todos, nextId: newNextId};
+    this.setState(newState, () => console.log(this.state));
+  }
+  
+  deleteTodo = todoId => {
+    const newTodos = this.state.todos.filter(todo => todo.id !== todoId);
+    this.setState({todos: newTodos}, () => console.log('Deleted'));
+  }
+  
+  toggleTodo = todoId => {
+    const todos = this.state.todos.map(todo => 
+      (todo.id === todoId)
+      ? {...todo, completed: !todo.completed}
+      : todo
+    )
+    this.setState({todos}, () => console.log('Toggled'));
+  }
+  
+  toggleAll = () => {
+    const allCompleted = this.state.todos.every(todo => todo.completed === true);
```

Comment:
It would be nice to have a Boolean value `allCompleted` in the state. It will prevent you from looping over an array many times and improve the performance